Model/Sales/MagentoOrderCollection.php

```diff
@@ -410,85 +498,21 @@ private function createShipment(Order $order)
     }
 
     /**
-     * Update all the tracks that made created via the API
+     * @param object $orderItem
      *
-     * @return $this
-     * @throws \Magento\Framework\Exception\LocalizedException
+     * @return string
      */
-    public function updateMagentoTrack()
+    private function getMultiStockInventory($orderItem): string
     {
-        /**
-         * @var $order        Order
-         * @var $shipment     Order\Shipment
-         * @var $magentoTrack Order\Shipment\Track
-         */
-        foreach ($this->getShipmentsCollection() as $shipment) {
-            $trackCollection = $shipment->getAllTracks();
-            foreach ($trackCollection as $magentoTrack) {
-                $myParcelTrack = $this->myParcelCollection->getConsignmentByApiId(
-                    $magentoTrack->getData('myparcel_consignment_id')
-                );
-
-                $magentoTrack->setData('myparcel_status', $myParcelTrack->getStatus());
+        $objectManager = ObjectManager::getInstance();
+        $sourceList    = $objectManager->get(Collection::class);
 
-                if ($myParcelTrack->getBarcode()) {
-                    $magentoTrack->setTrackNumber($myParcelTrack->getBarcode());
-                }
-                $magentoTrack->save();
+        foreach ($sourceList->load() as $sourceItemName) {
+            if ($sourceItemName->getName() === $orderItem->getSku()) {
+                return $sourceItemName->getSourceCode();
             }
         }
 
-        $this->updateGridByOrder();
-
-        return $this;
-    }
-
-    /**
-     * Update column track_status in sales_order_grid
-     *
-     * @return $this
-     * @throws \Magento\Framework\Exception\LocalizedException
-     */
-    public function updateGridByOrder()
-    {
-        if (empty($this->getOrders())) {
-            throw new LocalizedException(__('MagentoOrderCollection::order array is empty'));
-        }
-
-        /**
-         * @var Order $order
-         */
-        foreach ($this->getOrders() as $order) {
-            $aHtml = $this->getHtmlForGridColumns($order->getId());
-
-            if ($aHtml['track_status']) {
-                $order->setData('track_status', $aHtml['track_status']);
-            }
-            if ($aHtml['track_number']) {
-                $order->setData('track_number', $aHtml['track_number']);
-            }
-        }
-
-        $this->save();
-
-        return $this;
-    }
-
-    /**
-     * Check if there is 1 shipment in all orders
-     *
-     * @return bool
-     */
-    public function hasShipment()
-    {
-        /** @var $order Order */
-        /** @var Order\Shipment $shipment */
-        foreach ($this->getOrders() as $order) {
-            if ($order->hasShipments()) {
-                return true;
-            }
-        }
-
-        return false;
+        return '';
```

Comment:
Waarom lege string? Vind dat altijd een beetje vies, kan je niet string of null doen?

Reply:
done